### PR TITLE
[MODCONSKC-21] - Fix creation of entities and role entity

### PR DESCRIPTION
**Pull request by azizbekxm**
## Purpose
https://folio-org.atlassian.net/browse/MODCONSKC-21
Found issue in the creation of role entity where id is being set to null
image

## Approach
- Fixed issue
- Using builder to avoid any such cases in the future

## TODOS and Open Questions

- [ ] Check logging

## Pre-Merge Checklist:

Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added, or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do Rally stories exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail? Which endpoints/schemas changed, etc.
  - [ ] Do they have all the appropriate links to blocked/related issues?
- Are the Rally stories under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally, all the PRs involved in breaking changes would be merged on the same day to avoid breaking the folio-testing
environment. Communication is paramount if that is to be achieved, especially as the number of inter-module and
inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the
responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/consortia/service/impl/SharingRoleService.java [hunk 1]
```diff
@@ -95,12 +95,14 @@ protected void deleteSharingConfig(UUID roleId) {
...
-  protected SharingRoleEntity createSharingConfigEntityFromRequest(SharingRoleRequest sharingRoleRequest, String tenantId) {
-    SharingRoleEntity sharingRoleEntity = new SharingRoleEntity();
-    sharingRoleEntity.setId(UUID.randomUUID());
-    sharingRoleEntity.setRoleId(sharingRoleEntity.getRoleId());
-    sharingRoleEntity.setTenantId(tenantId);
-    return sharingRoleEntity;
+  protected SharingRoleEntity createSharingConfigEntityFromRequest(SharingRoleRequest request, String tenantId) {
+    return SharingRoleEntity.builder()
+      .id(UUID.randomUUID())
+      .roleId(request.getRoleId())
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/consortia/service/impl/SharingRoleService.java:101]: so this line was basically root cause of sharing issue, that roleId was not populated?

Comment (azizbekxm) [reply src/main/java/org/folio/consortia/service/impl/SharingRoleService.java:101]: yes  sharingRoleEntity.setRoleId(sharingRoleEntity.getRoleId()); roled is being populated using entity not request.

Comment (azizbekxm) [reply src/main/java/org/folio/consortia/service/impl/SharingRoleService.java:101]: image

File: src/main/java/org/folio/consortia/domain/entity/SharingPolicyEntity.java [hunk 2]
```diff
@@ -1,43 +1,32 @@
...
-
-  @Override
-  public int hashCode() {
-    return Objects.hash(id, policyId, tenantId);
-  }
 }
+
+
+
+
```
Comment (azizbekxm) [inline src/main/java/org/folio/consortia/domain/entity/SharingPolicyEntity.java:32]: will be remove soon

File: src/main/java/org/folio/consortia/domain/entity/SharingPolicyEntity.java [hunk 3]
```diff
@@ -1,43 +1,32 @@
...
+import org.folio.consortia.domain.entity.base.AuditableEntity;
 
-@Getter
-@Setter
-@ToString
+@Data
+@Builder
+@EqualsAndHashCode(callSuper = false)
 @RequiredArgsConstructor
+@AllArgsConstructor
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/consortia/domain/entity/SharingPolicyEntity.java:18]: Is it needed for tests?

Comment (azizbekxm) [reply src/main/java/org/folio/consortia/domain/entity/SharingPolicyEntity.java:18]: it is need for builder

File: src/test/java/org/folio/consortia/service/SharingPolicyServiceTest.java [hunk 4]
```diff
@@ -62,10 +63,11 @@ void shouldStartSharingPolicy() {
     var expectedPubRequestPut = createPublicationRequest(CONSORTIUM.getPolicyValue(), payload, HttpMethod.PUT)
       .tenants(Set.of(TENANT_ID_1))
       .url(request.getUrl() + "/" + request.getPolicyId());
```
Comment (BKadirkhodjaev) [inline src/test/java/org/folio/consortia/service/SharingPolicyServiceTest.java:65]: `String.format("%s/%s", request.getUrl(), request.getPolicyId())` can be nicer

File: src/main/java/org/folio/consortia/domain/entity/SharingRoleEntity.java [hunk 5]
```diff
@@ -3,21 +3,29 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Builder;
 import lombok.Data;
 import lombok.EqualsAndHashCode;
+import lombok.RequiredArgsConstructor;
 import org.folio.consortia.domain.entity.base.AuditableEntity;
 
 import java.util.UUID;
 
 @Data
```
Comment (BKadirkhodjaev) [inline src/main/java/org/folio/consortia/domain/entity/SharingRoleEntity.java:15]: `@RequiredArgsConstructor` can be removed here, `@Data` implicitly contains it (https://projectlombok.org/features/Data).

**Review by BKadirkhodjaev — APPROVED** [review]

File: src/main/java/org/folio/consortia/service/impl/SharingRoleService.java [hunk 6]
```diff
@@ -95,12 +95,14 @@ protected void deleteSharingConfig(UUID roleId) {
...
+      .tenantId(tenantId)
+      .isCapabilitiesShared(false)
+      .isCapabilitySetsShared(false)
```
[the remark this answers is not in the document]
Comment (azizbekxm) [reply src/main/java/org/folio/consortia/service/impl/SharingRoleService.java:104]: yes to inform other field to be transparent

File: src/main/java/org/folio/consortia/domain/entity/SharingRoleEntity.java (hunk 5 above)
Comment (azizbekxm) [reply src/main/java/org/folio/consortia/domain/entity/SharingRoleEntity.java:15]: Thanks, removed

File: src/test/java/org/folio/consortia/service/SharingPolicyServiceTest.java (hunk 4 above)
Comment (azizbekxm) [reply src/test/java/org/folio/consortia/service/SharingPolicyServiceTest.java:65]: Good idea, but From my perspective, current state is more simpler to read in tests and help to explain situtation more easier, so keeping it current format

**Review by SerhiiNosko — APPROVED** [review]

File: src/main/java/org/folio/consortia/domain/entity/SharingRoleEntity.java (hunk 5 above)
Comment (azizbekxm) [reply src/main/java/org/folio/consortia/domain/entity/SharingRoleEntity.java:15]: saw, error and to be on safe side reverted back requiredArgcontructor

**Review by yauhen-vavilkin — APPROVED** [review]

**Pull request merged**